### PR TITLE
Support `{scie.env.*}` referring to cmd env values.

### PR DESCRIPTION
**Pull request by jsirois**
This is needed by https://github.com/a-scie/lift for interpreter group
selection support and is otherwise a reasonable thing to expect to have
worked.

### CODE REVIEW

**Comment by jsirois** [comment]
My apologies in advance. This is rather large. That said, the concept is small and the bulk is tests.

File: examples/python/lift.linux-x86_64.json [hunk 1]
```diff
@@ -8,33 +8,30 @@
...
               "PEX_TOOLS": "1",
               "PEX_ROOT": "{scie.bindings}/pex_root",
-              "PEX_.*": null
+              "PEX_.*": null,
+              "=__PYTHON_3_8": "{cpython38}/python/bin/python3.8",
+              "=__PYTHON_3_9": "{cpython39}/python/bin/python3.9",
+              "=__PYTHON": "{scie.env.__PYTHON_3_{scie.env.PYTHON_MINOR=9}}"
             },
-            "exe": "{cpython}/python/bin/python3.9",
+            "exe": "{scie.env.__PYTHON}",
```
Comment (benjyw) [inline examples/python/lift.linux-x86_64.json:34]: Pretty neat!

**Review by benjyw — APPROVED** [review]
Looks good at the pseudocode level. My Rust is not good enough to comment in language-level detail.

File: examples/python/lift.linux-x86_64.json (hunk 1 above)
Comment (jsirois) [reply examples/python/lift.linux-x86_64.json:34]: All right, this took a good bit of fiddling to test out over in a-scie/lift to verify it meets all of scie-pants fairly advanced usage / needs of the scie-jump lift format. This enhancement does turn out to suffice.

**Pull request merged**